source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -168,7 +168,7 @@ def expand(self,unit):
 			except ValueError:
 				pass
 			try:
-				self._endOffset=text.index(u'\ufffc',oldEnd-self._startOffset)
+				self._endOffset=text.index(u'\ufffc', oldStart)

```

Comment:
I really don't think this is right. For 1, the text string is relative to self._startOffset. Therefore oldStart is simply not a valid index. It at least has to be relative to self._startOffset, such as oldStart-self._startOffset. 
To be clear, this code was written deliberately to only include text up to  the embedded object characters in either side. I.e. if positioned between two links, it should expand as far as possible without including the two links.
We can debate whether this was the right approach in the first place. There are historical reasons for this. And perhaps we should replace this code, and just  expand to the entire paragraph, and when collecting text actually recurse and grab all content from embedded objects and such. But, in that case, that would be an entirely different pr. Unless I'm completely mistaken in your approach, I would suggest closing this pr, and  looking at the problem more broadly.

Reply:
I understood the purpose of this code, but it introduced an offset as soon as the left side link were not at the start of the line. If I remember correctly, I wrote it like this because Firefox and Chrome don't behave exactly the same way here.
However, if you (@michaelDCurran) have a better idea or think that this code should be completely reworked (which I tend to think myself), I am OK with closing this PR in favor of another one.